packages/bpk-component-calendar/src/BpkCalendarContainer.tsx

```diff
@@ -40,20 +42,47 @@ import {
   startOfMonth,
 } from './date-utils';
 
+type Props = {
+  fixedWidth: boolean;
+  maxDate: Date;
+  minDate: Date;
+  onDateSelect?: (date: Date, newDate?: Date) => void;
+  onMonthChange?: (
+    event: UIEvent,
+    { month, source }: { month: Date; source: string },
+  ) => void;
+  selectionConfiguration: SelectionConfiguration;
+  initiallyFocusedDate: Date | null;
+};
+
+type CalendarProps<P> = P & Props;
+
+type State = {
+  preventKeyboardFocus: boolean;
+  focusedDate: Date;
+};
+
 /**
  * Updates the current focused date
  * @param {Object} currentProps current input properties
  * @param {Object} nextProps next input properties when component changes
  * @returns {Boolean} if the selected date has changed
  */
-const focusedDateHasChanged = (currentProps, nextProps) => {
+const focusedDateHasChanged = <T extends {}>(
+  currentProps: CalendarProps<T>,
+  nextProps: CalendarProps<T>,
+) => {
   const currentSelectConfig = currentProps.selectionConfiguration;
   const nextSelectConfig = nextProps.selectionConfiguration;
 
   const rawNextSelectedDate =
-    nextSelectConfig.date || nextSelectConfig.startDate;
+    nextSelectConfig.type === CALENDAR_SELECTION_TYPE.single
```

Comment:
this is the type for `SelectionConfiguration`:

export type SelectionConfigurationSingle = {
  type: typeof CALENDAR_SELECTION_TYPE.single;
  date: Date | null;
};

export type SelectionConfigurationRange = {
  type: typeof CALENDAR_SELECTION_TYPE.range;
  startDate: Date | null;
  endDate: Date | null;
};

export type SelectionConfiguration =
  | SelectionConfigurationSingle
  | SelectionConfigurationRange;